curriculum/challenges/english/25-front-end-development/review-html-accessibility/671a87a39b88245a579c2271.md

```diff
@@ -9,7 +9,46 @@ dashedName: review-html-accessibility
+
+## Accessibility Best Practices
+
+- **Proper heading level structure**: You should use proper heading levels to create a logical structure for your content. This helps screen readers and other assistive technologies understand the content of your website.
+- **Accessibility and Tables**: When using tables, you should use the `<th>` element to define header cells and the `<td>` element to define data cells. This helps screen readers and other assistive technologies understand the structure of the table.
+- **Importance for inputs to have an associated label**: You should use the `<label>` element to associate labels with form inputs. This helps screen readers and other assistive technologies understand the purpose of the input.
+- **Importance of good `alt` text**: You should use the `alt` attribute to provide a text alternative for images. This helps screen readers and other assistive technologies understand the content of the image.
+- **Importance of good link text**: You should use descriptive link text to help users understand the purpose of the link. This helps screen readers and other assistive technologies understand the content of the link.
+- **Best practices for making audio and video content accessible**: You should provide captions and transcripts for audio and video content to make it accessible to people with hearing impairments. You should also provide audio descriptions for video content to make it accessible to people with visual impairments.
+- **`tabindex` attribute**: used to specify the order in which elements should receive focus when using the tab key to navigate the website. This can help users with mobility impairments navigate the website more easily.
+- **`accesskey` attribute**: used to define a keyboard shortcut for an element. This can help users with mobility impairments navigate the website more easily.
+
+## WAI-ARIA, Roles, and Attributes
+
+- **WAI-ARIA**: a set of attributes that can be added to HTML elements to improve accessibility. It provides additional information to assistive technologies about the purpose and structure of the content.
+- **Aria roles**: a set of predefined roles that can be added to HTML elements to define their purpose. This helps screen readers and other assistive technologies understand the content of the website. Examples include `role="button"`, `role="dialog"`, and `role="navigation"`.
+- **`aria-label` and `aria-labelledby` attributes**: The `aria-label` attribute can be used to provide a text alternative for an element. The `aria-labelledby` attribute can be used to associate an element with another element that serves as its label. This helps screen readers and other assistive technologies understand the purpose of the element.
+- **`aria-hidden` attribute**: The `aria-hidden` attribute can be used to hide an element from screen readers. This can be useful for decorative elements that do not provide any meaningful content.
```

Comment:
```suggestion
- **`aria-hidden` attribute**: Used to hide an element from assistive technology such as screen readers. For example, this can be used to hide decorative images that do not provide any meaningful content.
```

Pure semantics here. While this is primarily used to hide content from screen reader users, it will also hide it from other types of assistive technology, basically anything that uses the accessibility tree API to communicate the contents on the page. If you think this distinction is too much or unnecessary, then I suggest it be changed to the following:

"Used to hide an element from screen reader users. For example, this can be used to hide decorative images that do not provide any meaningful content."